docs/src/user_docs/stdlib/math/u64.md

```diff
@@ -0,0 +1,69 @@
+| checked_xor |  Performs bitwise XOR of two unsigned 64 bit integers.<br /> The input values are assumed to be represented using 32 bit limbs, fails if they are not.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a XOR b. |
+| unchecked_shl |  Performs left shift of one unsigned 64-bit integer using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a << b mod 2^64.<br /> This takes 50 cycles. |
+| unchecked_shr |  Performs right shift of one unsigned 64-bit integer using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a >> b.<br /> This takes 66 cycles. |
+| overflowing_shl |  Performs left shift of one unsigned 64-bit integer preserving the overflow and<br /> using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [d_hi, d_lo, c_hi, c_lo, ...], where (d,c) = a << b, <br /> which d contains the bits shifted out.<br /> This takes 57 cycles. |
+| overflowing_shr |  Performs right shift of one unsigned 64-bit integer preserving the overflow and<br /> using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [d_hi, d_lo, c_hi, c_lo, ...], where c = a >> b, d = a << (64 - b).<br /> This takes 138 cycles. |
+| unchecked_rotl |  Performs left rotation of one unsigned 64-bit integer using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a << b mod 2^64.<br /> This takes 57 cycles. |
+| unchecked_rotr |  Performs right rotation of one unsigned 64-bit integer using the pow2 operation.<br /> The input value to be shifted is assumed to be represented using 32 bit limbs.<br /> The shift value is assumed to be in the range [0, 64).<br /> Stack transition looks as follows:<br /> [b, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a << b mod 2^64.<br /> This takes 62 cycles. |
```

Comment:
This documentation is outdated with the merge of #289 which made the pow2 operation more expensive. We can either update it now or wait until the new approach is done (depending on the expected timing of that approach with relation to the release of v0.2)